CMakeLists.txt

```diff
@@ -474,6 +474,12 @@ set(Boost_ADDITIONAL_VERSIONS
   "1.51.1" "1.51.0" "1.51"
   "1.50.1" "1.50.0" "1.50"
   "1.49.1" "1.49.0" "1.49")
+# Disable forced config-mode CMake search for Boost, which only imports targets
+# and does not set the variables that we need.
+#
+# TODO for the brave: transition all mlpack's CMake to 'target-based modern
+# CMake'.  Good luck!  You'll need it.
```

Comment:
Perhaps it's not that bad, you will learn a lot for sure.

Reply:
True... maybe I was a bit frustrated with it. :)  I'll open an issue for it and then maybe someone who is really brave might want to do it.  For just making mlpack build it's probably straightforward, but the mess that we have to do to generate `setup.py` for the Python bindings would be quite a lot of fighting, I think. :)

Reply:
Oh, I saw this afterward.